### ISSUE TITLE
easeTo() disturbs heading on Android.

### ISSUE THREAD

**Issue by dapetcu21** [mapbox/mapbox-gl-native#5559]
When setting the camera with `easeTo()` in a certain fashion (see below), if the current heading is not zero, the new heading will be altered randomly.

I confirmed this is happening with the example [here](https://www.mapbox.com/android-sdk/examples/).

**Platform:** Android
**Mapbox SDK version:** 4.1.0

### Steps to trigger behavior

 1. Initialize a map as per example.
 2. Rotate heading with finger
 3. Run the following on the press of a button:

```java
CameraPosition position = new CameraPosition.Builder(mapboxMap.getCameraPosition())
                                .target(new LatLng(40.72052634, -73.97686958312988))
                                .build(); // not changing heading in any way
CameraUpdate cameraUpdate = CameraUpdateFactory.newCameraPosition(position);
mapboxMap.easeCamera(cameraUpdate);
```

### Expected behavior

When pressing the button, the map view should pan to `40.72052634, -73.97686958312988`. Heading should remain unchanged.

### Actual behavior

Heading randomly jumps around when spamming the button.

**Comment by tobrun** [comment]
@dapetcu21 
thank you for reaching out, I was able to reproduce this with:

```java
                CameraPosition position = new CameraPosition.Builder(mapboxMap.getCameraPosition())
                        .target(new LatLng(40.72052634, -73.97686958312988))
                        .build(); // not changing heading in any way
                CameraUpdate cameraUpdate = CameraUpdateFactory.newCameraPosition(position);
                mapboxMap.easeCamera(cameraUpdate);
```

![ezgif com-video-to-gif 46](https://cloud.githubusercontent.com/assets/2151639/16559529/a4ffc9e8-41ed-11e6-984b-b19102fa37bd.gif)


and able to get a correct result with removing `mapboxMap.getCameraPostion()`:

```java
                CameraPosition position = new CameraPosition.Builder()
                        .target(new LatLng(40.72052634, -73.97686958312988))
                        .build(); // not changing heading in any way
                CameraUpdate cameraUpdate = CameraUpdateFactory.newCameraPosition(position);
                mapboxMap.easeCamera(cameraUpdate);
```

![ezgif com-video-to-gif 45](https://cloud.githubusercontent.com/assets/2151639/16559448/4acac144-41ed-11e6-81e1-68a1b73dc52d.gif)

I'm now looking why this is occurring, initial guess is that this is mismatching radiant versus degrees.

**Comment by tobrun** [comment]
While debugging, I'm noticing that the wrong value is coming from invalidating the camera position when the ease to animation has stopped.

**Comment by friedbunny** [comment]
I couldn’t reproduce this on iOS using a camera or via `setCenterCoordinate:animated:`. Here’s the camera equivalent:

```objc
// Use the existing camera (rather than creating a new one with zero’d values)
MGLMapCamera *camera = self.mapView.camera;
camera.centerCoordinate = CLLocationCoordinate2DMake(45, -122);
[self.mapView setCamera:camera animated:YES];
```

[This](https://github.com/mapbox/mapbox-gl-native/blob/2bf4e20052e827a4f69dd718ba659f66edccc6df/platform/ios/src/MGLMapView.mm#L2368-L2371) is the iOS implementation of `Map::getBearing()`:

```objc
- (CLLocationDirection)direction
{
    return mbgl::util::wrap(_mbglMap->getBearing(), 0., 360.);
}
```

**Comment by tobrun** [comment]
Recently revisited camera, final fixes merged with #6671. closing
